lib/msf/core/exploit/remote/kerberos/client.rb

```diff
@@ -207,11 +209,16 @@ def send_request_tgt_pkinit(options = {})
 
               pa_pk_as_rep = entry.decoded_value
               key = calculate_shared_key(pa_pk_as_rep, dh, dh_nonce, as_res.enc_part.etype)
-              return [Msf::Exploit::Remote::Kerberos::Model::TgtResponse.new(
+
+              return Msf::Exploit::Remote::Kerberos::Model::TgtResponse.new(
                 as_rep: as_res,
-                preauth_required: false,
-                decrypted_part: nil
-              ), key]
+                preauth_required: true,
```

Comment:
I'm not sure why `tgt_pkinit` doesn't do an initial AS_REQ to verify if preauth is required; But I believe this value should be true regardless.